prov/efa/src/efa_user_info.c

```diff
@@ -354,6 +355,36 @@ int efa_user_info_alter_rxr(struct fi_info *info, const struct fi_info *hints)
 
 			info->domain_attr->mr_mode |= FI_MR_HMEM;
 
+			/* TODO Loop over every HMEM interface and determine if we can support atomics */
+			support_atomic = true;
+			if (hmem_ops[FI_HMEM_CUDA].initialized && cuda_get_xfer_setting() != CUDA_XFER_ENABLED) {
+				/*
+				  HMEM_CUDA + ATOMICS require CUDA MEMCPY to be used.
+				  We are making an assumption here that the remote we are performing atomic
+				  operations on is the same as us. This might not be true for atomics over
+				  client/server model using EFA.  We do not currently have any users trying
+				  to do this, so we are not worried about this assumption yet.
+				*/
+				support_atomic = false;
+			}
+
+			if (!(hints->caps & FI_ATOMIC)) {
+				/*
+				   If the user doesn't explicity request FI_HMEM + FI_ATOMIC,
+				   don't claim support
+				*/
+				info->caps &= ~FI_ATOMIC;
+			} else if (!support_atomic) {
+				/*
+				 If the user explicity request FI_HMEM + FI_ATOMIC, but we do not support
+				 it for this type of hmem, do not claim atomic support
+				*/
+				EFA_WARN(FI_LOG_CORE,
+					"FI_HMEM + FI_ATOMIC support requested but not supported for HMEM type.\n");
+				return -FI_ENODATA;
+			} else {
+				info->caps &= FI_ATOMIC;
```

Comment:
This seems wrong. Looks like  everything in `info->caps` was cleared.

Reply:
Right, this should be `info->caps |= FI_ATOMIC` if you want to set the bit; `info->caps &= ~FI_ATOMIC` if you want to clear it.

Reply:
Switched to `info->caps |= FI_ATOMIC`